src/magicbox/MagicBox.ts

```diff
@@ -191,10 +189,9 @@ export class MagicBoxInstance {
     };
   }
 
-  public showSuggestion() {
-    this.suggestionsManager.mergeSuggestions(this.getSuggestions != null ? this.getSuggestions() : [], suggestions => {
-      this.updateSuggestion(suggestions);
-    });
+  public async showSuggestion() {
+    await this.suggestionsManager.receiveSuggestions(this.getSuggestions());
```

Comment:
What do you think of `addSuggesstions` or `pushSuggestions` or `processSuggestions` for the method name?

Reply:
I like those a lot better. I'll replace `showSuggestion` with `addSuggestions`.

Also, this made me notice that the method `clear` calls `showSuggestion` instead of `clearSuggestions`, so I'll see if it's safe to replace that too.